neqo-transport/src/streams.rs

```diff
@@ -20,6 +19,41 @@ use crate::{
 };
 use neqo_common::{qtrace, qwarn, Role};
 use std::{cell::RefCell, rc::Rc};
+use std::cmp::Ordering;
+
+pub type SendOrder = i64;
+
+#[derive(Copy, Clone)]
+pub struct StreamOrder {
+    pub sendorder: Option<SendOrder>,
+}
+
+// We want highest to lowest, with None being higher than any value
+impl Ord for StreamOrder {
+  fn cmp(&self, other: &Self) -> Ordering {
+    if self.sendorder.is_some() && other.sendorder.is_some() {
+      // We want reverse order (high to low) when both values are specified.
+      other.sendorder.cmp(&self.sendorder)
```

Comment:
The alternative (I keep coming up with these, sorry) might be better:

```suggestion
    if let (Some(s), Some(o)) = (&self.sendorder, &other.sendorder) {
      // We want reverse order (high to low) when both values are specified.
      o.cmp(s)
```